packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -58,31 +58,75 @@ type ConfigProps = {
   textColor: BaseTextProps['color']; // prop
   buttonSpacing: string; // raw value
   text?: string;
+  buttonColor: string; // raw value
+  buttonBorderColor: string; // raw value
+  hoverColor: string; // raw value
+  hoverBorderColor: string; // raw value
+  activeColor: string; // raw value
+  activeBorderColor: string; // raw value
+  focusColor: string; // raw value
+  focusBorderColor: string; // raw value
+  focusRingColor: string; // raw value
 };
 
 const getProps = ({
-  theme,
-  size,
-  children,
   buttonTypographyTokens,
+  children,
+  isDisabled,
+  size,
+  theme,
 }: {
+  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
+  children?: string;
+  isDisabled: boolean;
   theme: Theme;
   size: BaseButtonCommonProps['size'];
-  children?: string;
-  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
 }): ConfigProps => {
   const props: ConfigProps = {
     iconSize: 'medium',
     fontSize: buttonTypographyTokens.fonts.size.medium,
     lineHeight: buttonTypographyTokens.lineHeight.medium,
     buttonHeight: '40px',
     iconSpacing: makeSpacingSize(theme.spacing[2]),
-    iconColor: 'action.icon.primary.default',
-    textColor: 'action.text.primary.default',
+    iconColor: isDisabled ? 'action.icon.primary.disabled' : 'action.icon.primary.default',
+    textColor: isDisabled ? 'action.text.primary.disabled' : 'action.text.primary.default',
     buttonSpacing: `${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(
       theme.spacing[5],
     )} ${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(theme.spacing[5])}`,
     text: children?.trim(),
+    buttonColor: getIn(
```

Comment:
since the only differentiating prop is `disabled` here? then maybe abstract it out at a top-level and put it under the `if` condition and compute the disabled props, then just spread that object over here

Reply:
I've cleaned it up a bit, see if it makes sense